base/dataset.go

```diff
@@ -159,6 +159,8 @@ func ListDatasets(ctx context.Context, r repo.Repo, term string, limit, offset i
 				}
 				return nil, fmt.Errorf("error loading ref: %s, err: %s", ref.String(), err.Error())
 			}
+			ds.Peername = res[i].Peername
+			ds.Name = res[i].Name
```

Comment:
This through me for a loop, but I understand it now. Refs have a Peername and Name, and so do Datasets (as transient values), so we're copying them from the ref to the dataset before adding that to the result ref. The long term solution is getting rid of this duplication by switching off of reporef.Dataset, and instead using dsref.Ref and dataset.Dataset only.